### PR TITLE
Major refactoring

### PR DESCRIPTION
**Pull request by Klaas-**
* Add systemd_journal_create_directory boolean
* Add systemd_journal_storage option
* Add systemd_journal_compress option
* Add systemd_journal_max_retention_sec option
* Add systemd_journal_configuration to enable complex configuration without adding role parameters
* Refactor main.yml to deal with any journald option
* Refactor main.yml to use inifile instead of lineinfile
* Update Readme
* Update meta data

### CODE REVIEW

File: meta/main.yml [hunk 1]
```diff
@@ -1,15 +1,18 @@
 ---
 galaxy_info:
   author: groks
-  description: "enable the systemd journal and limit it's appetite for disk space"
+  description: "manage systemd journald configuration"
   license: BSD
-  min_ansible_version: 1.3
+  min_ansible_version: 2.1
```
Comment (martbhell) [inline meta/main.yml]: https://www.ansible.com/blog/loop-plays-past-present-future says that the loop: keyword was introduced in 2.5. I think we should set min version to 2.5

**Review by martbhell — CHANGES_REQUESTED** [review]
Hello and many thanks for this PR!

It looks very sweet.

I have in #3 added travis testing of the role. Would you be able to rebase the branch on master?
I hope that would auto-start another build on https://travis-ci.com/CSCfi/ansible-systemd-journal/builds

File: meta/main.yml (hunk 1 above)
Comment (Klaas-) [reply meta/main.yml]: yeah, I forgot about the loop :) fixed it in rebase

**Review by martbhell — APPROVED** [review]

**Comment by martbhell** [comment]
Thanks again :)

I have now also added some more testing (the role is idempotent) and at the end also cat journald.conf so we can see that it sets what we want: https://travis-ci.com/CSCfi/ansible-systemd-journal/jobs/184776160#L3771

**Comment by Klaas-** [comment]
@martbhell  no problem; thanks for integrating it so fast.

**Pull request merged**